app/controllers/api/v1/search_controller.rb

```diff
@@ -557,6 +574,27 @@ def self.escape_terms_for_regex(term_list:)
         /(#{escaped_terms.join('|')})/i
       end
 
+      # generate a Mongoid::Criteria object to perform a keyword/exact phrase search based on use case
+      # supports the following cases: :keyword (individual terms), :phrase (quoted phrases & keywords)
+      # and :inferred (converting a facet-based query to keywords)
+      # will scope the query based off of :base_studies, and include/exclude studies matching
+      # :accessions based on the :query_case (included by default, but excluded in :inferred to avoid duplicates)
+      def self.generate_mongo_query_by_case(terms:, base_studies:, accessions:, query_case:)
```

Comment:
hmmm.. I'm trying to think of a better word to use than 'case', since I at first thought this was referring to whether the search was case-sensitive. 'type'? 'term_type'? 'goal'? I'm not sure any of those are better though

Reply:
I landed on `generate_mongo_query_by_context`, and changed the last parameter to `query_context`.

Reply:
nice